### DISCUSSION TITLE
I created a little more advanced Bot

### DISCUSSION THREAD

**Discussion by heckmon** [Zain-ul-din/whatsapp-ai-bot#44] [Ideas]
Sir, inspired by you and some other developers, I created a little more advanced Bot which can be hosted in a server.

Please have a look in my repo😄

**Comment by Zain-ul-din** [comment]
cool! From where you added AI voice to your video it seems interesting to me :-)

**Reply by heckmon** [reply]
Capcut PC version has built in feature to convert subtitle (.srt) file to text to speech voice😁

**Reply by Zain-ul-din** [reply]
does Meta API key charges you?

**Reply by heckmon** [reply]
No Because I am using test number. It's free and unlimited. But if we need our Bot communicate with other people and groups, we need to set up a real mobile number for the bot. Which has 1000 chat/per month limit for free tier.

But this test number is free forever

**Reply by Zain-ul-din** [reply]
TY sharing your project I'll try to implement the following in my bot too you can contribute if you like to.

**Reply by Zain-ul-din** [reply]
Can we also deploy bot on the behalf of another person number like non-technical person.

**Reply by Zain-ul-din** [reply]
i mean if we've customers is it possible to sale it as a service?

**Reply by heckmon** [reply]
Meta doesn't allow that. A fresh whatsapp number or you must delete your existing whatsapp account inorder to make the bot spread to the world.

**Reply by heckmon** [reply]
Like I said earlier Vercel is a free hosting platform helps to host js apps for free. (I did some special configuration for my python app).
So it will be really helpful to make the bot always online.

**Reply by Zain-ul-din** [reply]
hmm

**Reply by heckmon** [reply]
We can't sell it. We need a registered mobile number if we want to make the bot available for everyone.(Or use the test number for personal use).
If you want to sell, the customers registered number can be used as a living body for the bot's soul

**Reply by heckmon** [reply]
Or you need to take a dedicated sim for the bot. And the bot can act as a official business number and manage your customer's issues in your absence

**Reply by Zain-ul-din** [reply]
I'm not uisng Meta API key in my bot. we don't need to register number this is just browser sending messages

**Reply by heckmon** [reply]
Then use test number. It's free forever and no need of any verification. But only the Bot is available for you as a personal bot.(Like the Bot in my video)

**Reply by heckmon** [reply]
For every meta account , meta gives a US number as test number (+1 73738 XXXX). That's the bot's number.
The test number only can communicate to its developer's number. So only we can access the bot as a personal bot. 

The real sim and number is only needed if the bot want to communicate globally to anyone or any groups

**Reply by Zain-ul-din** [reply]
Gotcha! What else do you know apart from that maybe we can colab

**Reply by heckmon** [reply]
I am not a developer 😅. I am just a student. I only know a little python.
I just learned about web APIs from YouTube and decided to make an API related project. Then I saw your repo and created a bot using python. I don't know anything about JS or typescript. But your app is entirely made from Typescript and JS. Then how can I colab without having any knowledge 😅

**Reply by Zain-ul-din** [reply]
cool! if you're a student go and get your student github pack it can be a valuable source for learning new skills.
